### DISCUSSION TITLE
What is hoisting exactly?

### DISCUSSION THREAD

**Discussion by jorgegomezcq** [squidfunk/mkdocs-material#6505] [Q&A] [answered]
Hi what is hoisting?

https://squidfunk.github.io/mkdocs-material/plugins/projects/#hoisting

It feels like the documentation does not introduce the definition of the term in this context. Correct me if I am wrong.

**Comment by squidfunk** [comment, marked as answer]
From this very documentation:

> Use this setting to enable or disable hoisting of themes files to the main project. If you disable this setting, each project receives a copy of the theme's files, which can be considered redundant [...] It's generally advisable to enable hoisting, as it yields faster deployments and faster loading of your project's sites, because the files are the same for all projects and can be deduplicated.

It ensures that there's only one copy of Material's theme files, i.e., CSS and JavaScript files. This improves caching and makes projects load faster, because the browser has to download only one copy of the same resources.

**Reply by jorgegomezcq** [reply]
What do you think of adding the following opening sentence in that section which introduces the term? The section starts by talking as if the reader already knows what hoisting is. I can see how, the reader can construct the definition of hoisting by putting the parts together but I think the below as an opening sentence might read more clearly.

> Hoisting ensures that there's only one copy of Material's theme files, i.e., CSS and JavaScript files. This improves caching and makes projects load projects faster.

**Reply by squidfunk** [reply]
Please note that this is the plugin's reference documentation. We're on our way restructuring our documentation to provide proper setup guides where we discuss those very features, which is a better place to explain it. It should be online shortly.
